### PR TITLE
Simplify project_iteration

### PR DESCRIPTION
**Pull request by ddufresne**
https://www.numigi.com/web#id=17184&action=298&model=project.task&view_type=form&menu_id=200

Use a many2one field to store the relation between task and parent project
instead of a many2many relationship.

The many2many was overkill and prevented from grouping tasks by parent project.

Also fix a bug. If a task's project has no parent and is not a parent project,
then the parent project for the task must be undefined.

### CODE REVIEW

**Comment by foutoucour** [comment]
@ddufresne could you share the link to the task ?

**Comment by ddufresne** [comment]
@foutoucour  here you go.

File: project_iteration/models/project_task.py [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 
     This field is used to navigate to the tasks from a parent project.
     """
 
     _inherit = 'project.task'
 
-    parent_project_ids = fields.Many2many(
-        'project.project', 'project_task_parent_rel', 'task_id', 'project_id',
-        string='Parent Project', compute='_compute_parent_project_ids', store=True)
+    parent_project_id = fields.Many2one(
```
Comment (foutoucour) [inline project_iteration/models/project_task.py:15]: Don't we need a migration script for client using already this module ?

File: project_task_subtask_same_project/models/project_task.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def action_subtask(self):
 
         context_vars_to_remove = (
             'search_default_project_id',
-            'search_default_parent_project_ids',  # defined by the module project_iteration
+            'search_default_parent_project_id',  # defined by the module project_iteration
```
Comment (foutoucour) [inline project_task_subtask_same_project/models/project_task.py:27]: I would version up the module. What do you think ?

**Review by foutoucour — CHANGES_REQUESTED** [review]

File: project_iteration/models/project_task.py (hunk 1 above)
Comment (ddufresne) [reply project_iteration/models/project_task.py:15]: Not for this. We are adding a new field and removing an old one.
These 2 cases are covered by a standard "odoo -u module".

File: project_task_subtask_same_project/models/project_task.py (hunk 2 above)
Comment (ddufresne) [reply project_task_subtask_same_project/models/project_task.py:27]: Of course

Comment (ddufresne) [reply project_task_subtask_same_project/models/project_task.py:27]: done

**Review by foutoucour — APPROVED** [review]

**Pull request merged**